lib/TaskProcessing/ContextChatSearchProvider.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\ContextChat\TaskProcessing;
+
+use OCA\ContextChat\AppInfo\Application;
+use OCA\ContextChat\Logger;
+use OCA\ContextChat\Service\LangRopeService;
+use OCA\ContextChat\Service\MetadataService;
+use OCA\ContextChat\Type\ScopeType;
+use OCP\IL10N;
+use OCP\TaskProcessing\ISynchronousProvider;
+
+class ContextChatSearchProvider implements ISynchronousProvider {
+
+	public function __construct(
+		private LangRopeService $langRopeService,
+		private IL10N $l10n,
+		private Logger $logger,
+		private MetadataService $metadataService,
+	) {
+	}
+
+	public function getId(): string {
+		return Application::APP_ID . '-context_chat_search';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('Nextcloud Assistant Context Chat Search Provider');
+	}
+
+	public function getTaskTypeId(): string {
+		return ContextChatSearchTaskType::ID;
+	}
+
+	public function getExpectedRuntime(): int {
+		return 120;
+	}
+
+	public function getInputShapeEnumValues(): array {
+		return [];
+	}
+
+	public function getInputShapeDefaults(): array {
+		return [];
+	}
+
+	public function getOptionalInputShape(): array {
+		return [];
+	}
+
+	public function getOptionalInputShapeEnumValues(): array {
+		return [];
+	}
+
+	public function getOptionalInputShapeDefaults(): array {
+		return [];
+	}
+
+	public function getOutputShapeEnumValues(): array {
+		return [];
+	}
+
+	public function getOptionalOutputShape(): array {
+		return [];
+	}
+
+	public function getOptionalOutputShapeEnumValues(): array {
+		return [];
+	}
+
+	/**
+	 * @inheritDoc
+	 * @return array{sources: list<string>}
+	 * @throws \RuntimeException
+	 */
+	public function process(?string $userId, array $input, callable $reportProgress): array {
+		if ($userId === null) {
+			throw new \RuntimeException('User ID is required to process the prompt.');
+		}
+
+		if (!isset($input['prompt']) || !is_string($input['prompt'])) {
+			throw new \RuntimeException('Invalid input, expected "prompt" key with string value');
+		}
+
+		if (
+			!isset($input['scopeType']) || !is_string($input['scopeType'])
+			|| !isset($input['scopeList']) || !is_array($input['scopeList'])
+			|| !isset($input['scopeListMeta']) || !is_string($input['scopeListMeta'])
+		) {
+			throw new \RuntimeException('Invalid input, expected "scopeType" key with string value, "scopeList" key with array value and "scopeListMeta" key with string value');
+		}
+
+		try {
+			ScopeType::validate($input['scopeType']);
+		} catch (\InvalidArgumentException $e) {
+			throw new \RuntimeException($e->getMessage(), intval($e->getCode()), $e);
+		}
+
+		// unscoped query
+		if ($input['scopeType'] === ScopeType::NONE) {
+			$response = $this->langRopeService->query($userId, $input['prompt']);
```

Comment:
`docSearch` method here instead of `query`

Reply:
Wow, sorry.